singularity-eos/eos/eos_mgusup.hpp

```diff
@@ -125,6 +125,47 @@ class MGUsup : public EosBase<MGUsup> {
   FillEos(Real &rho, Real &temp, Real &energy, Real &press, Real &cv, Real &bmod,
           const unsigned long output,
           Indexer_t &&lambda = static_cast<Real *>(nullptr)) const;
+
+  PORTABLE_FORCEINLINE_FUNCTION
+  Real MinimumDensity() const { return 10 * robust::EPS(); }
+
+  PORTABLE_FORCEINLINE_FUNCTION
+  Real MaximumDensity() const {
+    if (_s > 1) {
+      return 0.99 * robust::ratio(_s * _rho0, _s - 1);
+    } else { // for s <= 1, no maximum, but we need to pick something.
+      return 1e3 * _rho0;
+    }
+  }
+
+  template <typename Indexer_t = Real *>
+  PORTABLE_INLINE_FUNCTION void
+  DensityEnergyFromPressureTemperature(const Real press, const Real temp,
```

Comment:
Why not just use the base class version?

Reply:
Because of the maximum density, which is a runtime parameter for eos_mgusup but not the others. In the future we might want introspection for max density though. I'll add a comment explaining and a TODO to remove this when max density is available.

Reply:
Actually, we should just provide a max density that can be overwritten. I will add that.